File: Contest-1/C/Que4.c
int birthday(int s[], int d, int m, int n) {
    int prefixSum = 0;
    for (int i = 0; i < m; i++) {
        prefixSum += s[i];
    }
    int count = 0;

    if (prefixSum == d) {
        count++;
    }

    for (int i = 0; i < n; i++) {
        prefixSum -= s[i];
        if (i + m < n) {
            prefixSum += s[i + m];
        }
        if (prefixSum == d) {
            count++;
        }
    }
    
    printf("%d\n", count);
    return count;
}

File: Contest-1/Python/Que4.py
def birthday(s, d, m):
    prefixSum = 0
    n = len(s)
    
    for i in range(m):
        prefixSum += s[i]
    
    count = 0
    
    if prefixSum == d:
        count += 1
    
    for i in range(n):
        prefixSum -= s[i]
        if i + m < n:
            prefixSum += s[i + m]
        
        if prefixSum == d:
            count += 1
    
    print(count)
    return count
